detection_rules/rule.py

```diff
@@ -277,7 +277,7 @@ def skip_validate_note(self) -> bool:
         return os.environ.get('DR_BYPASS_NOTE_VALIDATION_AND_PARSE') is not None
 
     def validate_note(self):
-        if not (self.skip_validate_note or self.note):
+        if self.skip_validate_note or not self.note:
```

Comment:
Originally the check `if not (self.skip_validate_note or self.note)` was failing if DR_BYPASS_NOTE_VALIDATION_AND_PARSE was set. 

The condition evaluated to `if not (True or True)` which is `False` and did not skip validation.